scripts/mytop.sh

```diff
@@ -1225,32 +1371,42 @@ sub GetData()
 	$thread->{State}   ||= "";
 	$thread->{Progress} ||= 0;
 
-	## alter double hyphen comments so they don't break 
+        ## Alter double hyphen comments so they don't break
 	## the query when newlines are removed - http://freshmeat.net/users/jerjones
-	$thread->{Info} =~ s~\s--(.*)$~ /* $1 */ ~mg; 
+        $thread->{Info} =~ s~\s--(.*)$~ /* $1 */ ~mg;
 
         ## Normalize spaces -- mostly disabled for now.  This can
         ## break EXPLAIN if you try to explain a mangled query.  It
         ## may be re-enabled later as an option.
 
-        ## leading space removal
+        ## Replace newlines and carriage returns with a space
+        $thread->{Info} =~ tr/\n\r/ /;
+
+        ## Leading space removal
         $thread->{Info} =~ s/^\s*//;
 
-        if (1)
-        {
-            ## remove newlines and carriage returns
-            $thread->{Info} =~ s/[\n\r]//g;
+        ## Strip non printing control symbols
+        $thread->{Info} =~ tr/[[:cntrl:]]//;
 
-            ## collpase whitespace
-            $thread->{Info} =~ s/\s+/ /g;
-        }
+        ## Collpase whitespace
+        $thread->{Info} =~ s/\s+/ /g;
+
+        ## Trailing space removal
+        $thread->{Info} =~ s/\s$//;
 
-        ## stow it in the cache
+        ## Put the first letter of the query uppercase for a better readability with long State strings
+        $thread->{Info} = ucfirst $thread->{Info};
 
+        ## Stow it in the cache
         $qcache{$thread->{Id}}  = $thread->{Info};
         $dbcache{$thread->{Id}} = $thread->{db};
         $ucache{$thread->{Id}}  = $thread->{User};
 
+        ## If Progress information is available and a multi-stage query is running, the actual stage and the total number of stages of the thread are shown at the beginning of the State column
```

Comment:
On `information_schema.processlist` since MariaDB 5.3 there are 2 columns `STAGE` and `MAX_STAGE`.

In normal queries without any stage, the `%` (progress) column is directly taking the value from the `PROGRESS` column but on some operations such as an ALTER ENGINE that has multiple stages, the `PERCENT` column is relative to the actual `STAGE`, meaning that for a query with 5 stages the `PERCENT` columns might go from 100 to 0 multiple times and this not really be a good indication of the actuel progress.

With this change it might for example show `1/5 Doing Something` on the State column and `95%` on the `%` column, meaning that its a 95% of the first stage and not 95% of the overall query execution.

Reply:
I understand logic, but still I couldn't reproduce this to verify and see how it behaves. Can you show me the output or give me a test case ? 
As I stated for color changes something happened with your patch and I couldn't find what is changing this, please see bellow: 
![image](https://user-images.githubusercontent.com/8207668/65037544-3c120080-d94e-11e9-9db2-09d621aba81a.png)
Note also that your patch is missing information about the build type  (debug, release).
80 char rule.